client/state/data-layer/wpcom/read/following/mine/new/index.js

```diff
@@ -40,16 +40,17 @@ export function receiveFollow( store, action, next, response ) {
 			)
 		);
 	} else {
-		followError( store, action, next );
+		followError( store, action, next, response );
 	}
 }
 
-export function followError( { dispatch }, action, next ) {
+export function followError( { dispatch }, action, next, response ) {
 	dispatch(
 		errorNotice(
 			translate( 'Sorry, there was a problem following that site. Please try again.' )
 		)
 	);
+	dispatch( recordFollowError( action.payload.feedUrl, response ) );
```

Comment:
I think I would prefer to hand the action creator the error info directly, instead of having the action creator have to know about the shape of the error response.